### DISCUSSION TITLE
[Help]: Embla Carousel Infinite Scroll Backwards

### DISCUSSION THREAD

**Discussion by filipe-a-rosa-alb** [davidjerleke/embla-carousel#1075] [Help]
### Summary

Hello community,

I have seen you have an example of infinite scroll in react scrolling forward, is it possible to do the same behaviour when scrolling backwards? There is any example out there with the infinite scroll backwards behaviour?

Thanks in advance,
Filipe

### If applicable, which variants of Embla Carousel are relevant to this question?

- [x] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
Hi @filipe-a-rosa-alb,

Thanks for your question. I'm not sure what you want to achieve? Do you mean infinite scroll with a RTL setup? Because by default, a carousel can only scroll forward unless `loop` is `true`?

**Reply by filipe-a-rosa-alb** [reply]
Hi @davidjerleke, in your example of infinite scroll, the infinite scroll is only working when scrolling forward. I would like to achieve the same behavior scroll backwards aswell, which means when reaching the index 0 of carousel, load more data at the beginning of array and be able to scroll to the new data loaded at the start of array.

I have an example where I tried to achieve the behavior of infinite scroll backwards, with no success. Here is the code sandbox link where you can scroll backwards and see the wanted behavior. In this example it's visually working fine but for what i can see internally the indexes are not okay.

Codesandbox: https://codesandbox.io/p/sandbox/embla-carousel-infinite-scroll-react-forked-r33hfz

**Reply by davidjerleke** [reply]
@filipe-a-rosa-alb unfortunately I'm not planning to add this feature. It's very confusing UX because it leads to disorientation. Users will have no idea of where they are.

**Reply by filipe-a-rosa-alb** [reply]
Thank you for your quick answer. Is there any chance to possibly achieve this feature by changing your example of infinite scroll 
online?

**Reply by davidjerleke** [reply]
@filipe-a-rosa-alb unfortunately I don't have time to look into it. As much as I would like to help everybody with their use cases my spare time is very limited. I need to focus on more general things rather than specific use cases. Thanks for understanding!

I'll leave this question open for a period of time though and maybe someone from the community will pick it up.

**Comment by siimsams** [comment]
Hey!

I also have this requirement. I'm building a calendar which will be scrollable both directions and new days will be loaded in as the user scrolls. The problem is that the way I got it working currently will kill the animation that's going on. Is there a way to get this working?

I also tried to achieve similar thing with another known swiper library but so far it has issues with maintaining order. The most difficult part of this problem seems to be the order maintaining especially if swiping to the left.
